web/source/keyboards/kmwkeyboards.ts

```diff
@@ -975,18 +1002,27 @@ namespace com.keyman.keyboards {
                 }
               }
             } else { // Single language element
-              tEntry = new CloudRequestEntry(x[i]['id'], x[i]['languages'][j]['id']);
+              tEntry = new CloudRequestEntry(x[i]['id'], x[i]['languages']['id']);
```

Comment:
`[j]` undefined

Reply:
So was this fixing an earlier bug? Can we put in a unit test if so?

Reply:
It turns out current unit tests don't exercise this portion of code because the unit tests run independent of API calls (keyboard files provided in the stub).

I spent some time trying to devise a unit test with this stub
```
      // Don't include filename to enforce checking languages Object in keyboard stub  var stub = {
        "id": "sil_euro_latin",
        "name": "EuroLatin (SIL)",
        "languages": {
          "id": "af"
        }
      };
```

and verified it exercises this part of the code. But since the test needs an api.keyman.com call to get the filename, we'll just consider this tested in the promise-api test page.

Reply:
Hmm.  That filename-less stub sounds like it may be worth keeping around as part of the manual-test page.  Add a button somewhere to facilitate registering the partial-stub, and make it clear why the button exists.

Reply:
Added button 4 to promise-api test page for this